twine/__main__.py

```diff
@@ -20,12 +20,25 @@
 from twine import cli
 from twine import exceptions
 
+try:
+    import colorama
+except Exception:
+    colorama = None
+
 
 def main() -> Any:
     try:
+        if colorama:
+            colorama.init()
         return cli.dispatch(sys.argv[1:])
     except (exceptions.TwineException, requests.HTTPError) as exc:
-        return "{}: {}".format(exc.__class__.__name__, exc.args[0])
+        pre_style, post_style = "", ""
+        if colorama:
+            pre_style, post_style = colorama.Fore.RED, colorama.Style.RESET_ALL
+
+        return "{}{}: {}{}".format(
+            pre_style, exc.__class__.__name__, exc.args[0], post_style,
+        )
```

Comment:
As noted by @sigmavirus24 in https://github.com/pypa/twine/issues/511#issuecomment-616133355, I'm curious what happens when the output isn't going straight to a terminal, e.g. `twine upload dist/* 2> errors.txt`. According do the [colorama docs](https://github.com/tartley/colorama#init-keyword-args):

> The default behaviour is to strip if on Windows or if output is redirected (not a tty)

However, I wonder if that still works since the colorized string is being passed to `sys.exit`, instead of `print` or `sys.stderr.write`.

Have you tried that manually? Off the top of my head, an automated test might involve using `subprocess` to run the `twine` command directly.

Reply:
I just tested this on my mac and the default behavior you cited holds true, no difference in behavior between this branch and master.

However, I found [this issue on pip (\#6354)](https://github.com/pypa/pip/issues/6354) where the coloroma color doesn't reset properly on Windows when redirecting pip output. See also these two colorama issues [\#200](https://github.com/tartley/colorama/issues/200) and [\#218](https://github.com/tartley/colorama/issues/218) which discuss issues on Windows when redirecting output. I have some ideas about how we can get around these issues if they arise, but I'd like to test that they're existing problems for us in the first place.

With regards to testing, I can go ahead and start writing some to test this behavior automatically, however at this point we're only automatically testing against Linux with Travis CI. As I mentioned above, I'm also working on #650 to test against Windows and macOS as well, so assuming I can get that up and running soon, that should help motivate these tests.

Reply:
> I just tested this on my mac and the default behavior you cited holds true, no difference in behavior between this branch and master.

Likewise.

Are you still looking into this on Windows?

I think this also relates to the [comment](https://github.com/pypa/twine/pull/649/files#r434808139) in `test_main.py`